api/server/router/container/container_routes.go

```diff
@@ -386,10 +385,6 @@ func (s *containerRouter) deleteContainers(ctx context.Context, w http.ResponseW
 	}
 
 	if err := s.backend.ContainerRm(name, config); err != nil {
-		// Force a 404 for the empty string
```

Comment:
Since if name is empty, in the implementation of `ContainerRm`, `getContainer` will take into empty string into consideration, and return an error 
```
func (daemon *Daemon) GetContainer(prefixOrName string) (*container.Container, error) {
	if len(prefixOrName) == 0 {
		return nil, errors.NewBadRequestError(fmt.Errorf("No container name or ID supplied"))
	}
        ....
}
```

So if the string is empty, will return an error of `fmt.Errorf("No container name or ID supplied")` instead of `ErrEmptyPrefix = errors.New("Prefix can't be empty")`. So remove the code above.